specification/saas/resource-manager/Microsoft.SaaS/preview/2018-03-01-beta/saas.json

```diff
@@ -356,6 +356,445 @@
+                "description": "The URL to check the status of the asynchronous operation.",
+                "type": "string"
+              },
+              "Retry-After": {
+                "description": "The amount of delay to use while the status of the operation is checked. The value is expressed in seconds.",
+                "format": "int32",
+                "type": "integer"
+              }
+            }
+          },
+          "204": {
+            "description": "Operation completed successfully."
+          },
+          "default": {
+            "description": "A response indicating an error.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create SaaS resource": {
+            "$ref": "./examples/saasSubscriptionLevel/SaasPatch.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "SubscriptionLevelSaaSResources"
+        ],
+        "operationId": "SaasSubscriptionLevel_Delete",
+        "description": "Deletes the specified SaaS.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Operation completed successfully."
+          },
+          "202": {
+            "description": "Accepted and the operation will complete asynchronously.",
+            "headers": {
+              "Location": {
+                "description": "The URL to check the status of the asynchronous operation.",
+                "type": "string"
+              },
+              "Retry-After": {
+                "description": "The amount of delay to use while the status of the operation is checked. The value is expressed in seconds.",
+                "format": "int32",
+                "type": "integer"
+              }
+            }
+          },
+          "204": {
+            "description": "Operation completed successfully."
+          },
+          "default": {
+            "description": "A response indicating an error.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Delete Subscription Level SaaS resource": {
+            "$ref": "./examples/saasSubscriptionLevel/SaasDelete.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.SaaS/resources/{resourceName}/unsubscribe": {
+      "patch": {
```

Comment:
>patch [](start = 7, length = 5)

An imperative action like `unsubscribe` should be using the POST verb in ARM. Patch in ARM is reserved for resources.

Reply:
Also how does one re-subscribe if this is called in error?

---
In reply to: [551677841](https://github.com/Azure/azure-rest-api-specs/pull/11903#discussion_r551677841) [](ancestors = 551677841)

Reply:
Or is this supposed to behave like delete?

---
In reply to: [551678647](https://github.com/Azure/azure-rest-api-specs/pull/11903#discussion_r551678647) [](ancestors = 551678647,551677841)

Reply:
unsubscribe is like delete, just that we stop billing etc. but not delete the ARM resource, so the users will still be able to see the resource and it's state.
Will change to POST